src/renderer.zig

```diff
@@ -606,6 +611,90 @@ pub const MenuBackGround = struct {
 	}
 };
 
+pub const Skybox = struct {
+	var shader: Shader = undefined;
+	var uniforms: struct {
+		time: c_int,
+		altitude: c_int,
+		lightDir: c_int,
+		invLightDir: c_int,
+		viewMatrix: c_int,
+		projectionMatrix: c_int,
+	} = undefined;
+
+	var vao: c_uint = undefined;
+	var vbos: [2]c_uint = undefined;
+
+	fn init() void {
+		shader = Shader.initAndGetUniforms("assets/cubyz/shaders/skybox/vertex.vs", "assets/cubyz/shaders/skybox/fragment.fs", "", &uniforms);
+		shader.bind();
+		
+		const rawData = [_]f32 {
+			-1, -1, -1,
+			1, -1, -1,
+			1, 1, -1,
+			-1, 1, -1,
+			-1, -1, 1,
+			1, -1, 1,
+			1, 1, 1,
+			-1, 1, 1
+		};
+
+		const indices = [_]c_int {
+			0, 3, 1, 1, 3, 2,
+			5, 6, 4, 4, 6, 7,
+			3, 7, 2, 2, 7, 6,
+			1, 5, 0, 0, 5, 4,
+			4, 7, 0, 0, 7, 3,
+			1, 2, 5, 5, 2, 6,
+		};
+
+		c.glGenVertexArrays(1, &vao);
+		c.glBindVertexArray(vao);
+		c.glGenBuffers(2, &vbos);
+		c.glBindBuffer(c.GL_ARRAY_BUFFER, vbos[0]);
+		c.glBufferData(c.GL_ARRAY_BUFFER, @intCast(rawData.len*@sizeOf(f32)), &rawData, c.GL_STATIC_DRAW);
+		c.glVertexAttribPointer(0, 3, c.GL_FLOAT, c.GL_FALSE, 3*@sizeOf(f32), null);
+		c.glEnableVertexAttribArray(0);
+		c.glBindBuffer(c.GL_ELEMENT_ARRAY_BUFFER, vbos[1]);
+		c.glBufferData(c.GL_ELEMENT_ARRAY_BUFFER, @intCast(indices.len*@sizeOf(c_int)), &indices, c.GL_STATIC_DRAW);
+	}
+
+	pub fn deinit() void {
+		shader.deinit();
+		c.glDeleteVertexArrays(1, &vao);
+		c.glDeleteBuffers(2, &vbos);
+	}
+
+	pub fn render() void {
+		c.glDisable(c.GL_CULL_FACE);
+		c.glDisable(c.GL_DEPTH_TEST);
+
+		const viewMatrix = game.camera.viewMatrix;
+		shader.bind();
+
+		const time = @as(f32, @floatFromInt(@mod(main.game.world.?.gameTime.load(.monotonic), main.game.World.dayCycle))) / (main.game.World.dayCycle);
+
+		const lightMatrix = Mat4f.rotationY(-std.math.pi / 2.0 - time * std.math.pi * 2);
+
+		const invLightMatrix = Mat4f.rotationY(time * std.math.pi * 2 + std.math.pi / 2.0);
+
+		c.glUniform1f(uniforms.time, time);
+		c.glUniform1f(uniforms.altitude, @floatCast(game.Player.super.pos[2]));
+		c.glUniformMatrix4fv(uniforms.lightDir, 1, c.GL_TRUE, @ptrCast(&lightMatrix));
+		c.glUniformMatrix4fv(uniforms.invLightDir, 1, c.GL_TRUE, @ptrCast(&invLightMatrix));
+		
+		c.glUniformMatrix4fv(uniforms.viewMatrix, 1, c.GL_TRUE, @ptrCast(&viewMatrix));
+		c.glUniformMatrix4fv(uniforms.projectionMatrix, 1, c.GL_TRUE, @ptrCast(&game.projectionMatrix));
+
+		c.glBindVertexArray(vao);
+		c.glDrawElements(c.GL_TRIANGLES, 36, c.GL_UNSIGNED_INT, null);
+
+		c.glEnable(c.GL_CULL_FACE);
+		c.glEnable(c.GL_DEPTH_TEST);
```

Comment:
Please encapsulate this function in `gpu_performance_measuring.start/stopQuery` of a new type, so it shows up in the GPU perfomance debug menu (F5).